### ISSUE TITLE
Megatorch not blocking Silent Gems 3 spawns 1.15.2 MC (Mod Compatibility)

### ISSUE THREAD

**Issue by Zeure** [Xalcon/TorchMaster#121]
Hello,

MC version 1.15.2

Issue: hostile mobs from the Silent Gems 3 mod are not being blocked from spawning within the radius of the megatorch.  I changed the config up to a higher value to test it out and have not been able to keep them from spawning.  I'm playing on the Endless Frontier modpack from the twitch launcher.  I haven't changed anything else in the configs.  I'm not sure if you need reports from this version of MC, but I thought it couldn't hurt.

Thank you.

**Comment by Xalcon** [comment]
can you run the command `/torchmaster entitydump` and give me your latest.log afterwards? It seems silent gems either spawns those entities differently than minecraft itself or they are not considered hostile.

**Comment by McKitten** [comment]
I was under the impression that this was by design as slimes are not affected by light levels. 
Silent Gems has its own utility to reduce slimes & wisps from spawning.

**Comment by Akjosch** [comment]
Not the OP, but I was looking into something similar recently, and it tools like it should work in reasonably current versions of both mods for 1.15.2.  Technical details:

* MC TechColonies modpack version 3.5, TorchMaster 2.1.1-rc0, Silent's Gems 1.15.2-4.6.3+55

* "/torchmaster entitydump" includes all the Silent's Gems mods you'd expect it to:
```
[28Jul2020 10:33:56.343] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:chaos_wisp
[28Jul2020 10:33:56.343] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:ender_slime
[28Jul2020 10:33:56.344] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:water_wisp
[28Jul2020 10:33:56.345] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:corrupted_slime
[28Jul2020 10:33:56.345] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:ice_wisp
[28Jul2020 10:33:56.345] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:lightning_wisp
[28Jul2020 10:33:56.345] [Server thread/INFO] [net.xalcon.torchmaster.Torchmaster/]:   silentgems:fire_wisp
```
(It also includes the related Silent's Gears' "Slingshot Projectile", which I'm not sure it should - but that might be simply that mod deriving a projectile from an aggressive mob entity base class ...)

* Relevant code parts in Silent's Gems:
-- Standard Forge event check to see if spawning isn't prevented: https://github.com/SilentChaos512/SilentGems/blob/1.15/src/main/java/net/silentchaos512/gems/world/spawner/CorruptedSlimeSpawner.java#L65
-- Corrupted Slime deriving from `net.minecraft.entity.monster.SlimeEntity`: https://github.com/SilentChaos512/SilentGems/blob/1.15/src/main/java/net/silentchaos512/gems/entity/CorruptedSlimeEntity.java#L15

**Comment by Xalcon** [comment]
* Are those entities listed under either the dreadlamp or the mega torch? Or just as "available"?
* Have you tried adding them explicitly to the megatorch via the config?

**Comment by Akjosch** [comment]
Those entities are listed for the mega torch. And I didn't add anything to the config (it's not *my* mod pack anyway), it just seems to work with reasonably new versions of both mods.

Full `torchmaster.toml` file in that modpack looks like this:
```
[General]
	#The maximum amount of invisble lights a feral flare lantern can place. Set to 0 to disable light placement.
	#Warning: Setting this value too high in conjunction with the feralFlareMinLightLevel and Radius can lead to world corruption!
	#(Badly compressed packet error)
	#Range: 0 ~ 32767
	feralFlareLanternLightCountHardcap = 255
	#Same as the mega torch block list override, just for the dread lamp
	#Block squid: +minecraft:squid
	#Allow pigs: -minecraft:pig
	dreadLampEntityBlockListOverrides = []
	#The radius of the mega torch in each direction (cube) with the torch at its center
	#Range: > 0
	megaTorchRadius = 64
	#Show additional information in the tooltip of certain items and blocks
	beginnerTooltips = true
	#Use this setting to override the internal lists for entity blocking
	#You can use this to block more entities or even allow certain entities to still spawn
	#The + prefix will add the entity to the list, effectivly denying its spawns
	#The - prefix will remove the entity from the list (if necessary), effectivly allowing its spawns
	#Note: Each entry needs to be put in quotes! Multiple Entries should be separated by comma.
	#Block zombies: "+minecraft:zombie"
	#Allow creepers: "-minecraft:creeper"
	megaTorchEntityBlockListOverrides = []
	#Durability of the frozen pearl. Each removed light will remove one charge from the pearl. Set to 0 to disable durability
	#Range: 0 ~ 32767
	frozenPearlDurability = 1024
	#Controls how often the flare should try to place lights. 1 means every tick, 10 every 10th tick, etc
	#Range: > 1
	feralFlareTickRate = 5
	#If this setting is enabled, the mega torch will block all natural spawn attempts from Lycanites Mobs in its radius
	lycanitesMobsBlockAll = true
	#The radius in which the feral flare should try to place lights
	#Range: 1 ~ 127
	feralFlareRadius = 16
	#The target minimum light level to place lights for
	#Range: 0 ~ 15
	feralFlareMinLightLevel = 15
	#By default, mega torches only block natural spawns (i.e. from low light levels). Setting this to false will also block spawns from spawners
	blockOnlyNaturalSpawns = true
	#The radius of the dread lamp in each direction (cube) with the torch at its center
	#Range: > 0
	dreadLampRadius = 64

```

**Comment by Xalcon** [comment]
/edit
I'm stupid - i didnt read everything you wrote. 
I hope I can find the time to check this myself soon but I don't have a working 1.15 development environment at the moment.

**Comment by Xalcon** [comment]
I've checked locally with just Torchmaster and Silent Gems and I'm unable to replicate the issue. I've checked my code and noticed some missing features from earlier versions - specifically the more aggressive spawn prevention option. While I'm not sure if this will fix the issue, its definitely something that should be in the mod and I hope it works for you. 

Generally, when the CheckSpawn even is running, I check if any other mod in the chain already set the spawn result to "Allow" rather than "Default". The aggressiveSpawnCheck config setting will override this check but it is disabled by default to avoid issues with other spawn controlling mods.

This feature will be available in Torchmaster 2.2.2 for MC 1.15 and 2.3.1 for MC 1.16. Please let me know if enabling this option changes anything for you.

**Comment by Xalcon** [comment]
Closed due to inactivity
